src/connection/eagerlyConnect.ts

```diff
@@ -43,9 +45,14 @@ if (selectedWallet) {
   if (selectedConnection) {
     connectionReady = connect(selectedConnection.connector, selectedWallet)
       .then((connected) => {
-        if (!connected) {
-          // only clear the persisted wallet type if it failed to connect.
-          localStorage.removeItem(selectedWalletKey)
+        if (!connected) throw new FailedToConnect()
+      })
+      .catch((error) => {
+        // Clear the persisted wallet type if it failed to connect.
+        localStorage.removeItem(selectedWalletKey)
+        // Log it if it threw an unknown error.
+        if (!(error instanceof FailedToConnect)) {
+          console.error(error)
```

Comment:
do we do anything with console logged errors? Should we instead log it to Sentry?

Reply:
A user may just be denying a connection. I think this is the same behavior as before.